File: ObjectQueue/C#/Student.cs
class Student {
    public int Registration {get ; set;}
    public string Name {get ; set;}

    public Student(int Registration, string Name) {
        this.Registration = Registration;
        this.Name = Name;
    }

    
    public override string ToString() {
        return "Aluno [Registration=" + Registration + ", Name=" + Name + "]";
    }

    
    public override bool Equals(object obj) {
     
        
        Student other = (Student) obj;
        if (Registration != other.Registration)
            return false;
        if (Name == null) {
            if (other.Name != null)
                return false;
        } else if (!Name.Equals(other.Name))
            return false;
        return true;
    }
}

File: ObjectQueue/Java/Student.java
package ObjectQueue.Java;

public class Student {
    public int registration;
    public String name;

    public Student(int registration, String name) {
        this.registration = registration;
        this.name = name;
    }

    @Override
    public String toString() {
        return "Aluno [registration=" + registration + ", name=" + name + "]";
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        Student other = (Student) obj;
        if (registration != other.registration)
            return false;
        if (name == null) {
            if (other.name != null)
                return false;
        } else if (!name.equals(other.name))
            return false;
        return true;
    }

    
}
 
    

